go-controller/pkg/ovn/egressgw.go

```diff
@@ -474,33 +471,35 @@ func (oc *Controller) deleteGWRoutesForPod(name ktypes.NamespacedName, podIPNets
 	defer routeInfo.Unlock()
 
 	for _, podIPNet := range podIPNets {
-		pod := podIPNet.IP.String()
-		if gwToGr, ok := routeInfo.podExternalRoutes[pod]; ok {
+		podIP := podIPNet.IP.String()
+		if gwToGr, ok := routeInfo.podExternalRoutes[podIP]; ok {
 			if len(gwToGr) == 0 {
-				delete(routeInfo.podExternalRoutes, pod)
+				delete(routeInfo.podExternalRoutes, podIP)
 				continue
 			}
-			mask := GetIPFullMask(pod)
 			for gw, gr := range gwToGr {
-				node := util.GetWorkerFromGatewayRouter(gr)
-				portPrefix, err := oc.extSwitchPrefix(node)
-				if err != nil {
-					klog.Infof("Failed to find ext switch prefix for %s %v", node, err)
-					continue
-				}
-				if err := oc.deleteLogicalRouterStaticRoute(pod, mask, gw, gr); err != nil {
+				mask := GetIPFullMask(podIP)
```

Comment:
this should be kept outside of the loop so we dont keep doing it for each gr

Reply:
Done.